### ISSUE TITLE
DDQN compute target

### ISSUE THREAD

**Issue by dai-dao** [flyyufelix/C51-DDQN-Keras#1]
Hey, nice implementation and blog post. I have a question about the code, for these 2 lines:

```
z = self.model.predict(next_states) # Return a list [32x51, 32x51, 32x51]
z_ = self.model.predict(next_states) # Return a list [32x51, 32x51, 32x51]
```

where z_ is used to compute the target, and z is used to draw the max value action. But shouldn't it be:

```
z_ = self.target_model.predict(next_states) # Return a list [32x51, 32x51, 32x51]
```
??

**Comment by flyyufelix** [comment]
Thanks for pointing that out. Yes you are right, it should be target_model instead of model. I've just uploaded the latest version to github to reflect the change.

**Comment by dai-dao** [comment]
It's all good. I have also implemented c51 with Gluon - Mxnet, but it's not learning anything, and I don't really know why. It's nice that you were able to get some good results. 

One other question for you is, why are you using the Categorial cross entropy loss? In the keras documentation it says that it only applies to labels such as [0, 0, 1], not a probability distribution.

**Comment by flyyufelix** [comment]
No, the "labels" do not need to be binary, it can be a probability distribution. You can look up the source code of categorical cross entropy loss https://github.com/fchollet/keras/blob/master/keras/losses.py

The implementation also didn't work in the beginning. It definitely took me some time to get it to work :)

**Comment by dai-dao** [comment]
Thanks for the feedback. Cheers :)
